src/solution.c

```diff
@@ -417,20 +418,19 @@ static msg_t solution_thread(void *arg)
     static u8 n_ready_old = 0;
     u64 nav_tc = nap_timing_count();
     static navigation_measurement_t nav_meas[MAX_CHANNELS];
-
-    const channel_measurement_t *p_meas[n_ready];
-    navigation_measurement_t *p_nav_meas[n_ready];
-    const ephemeris_t *p_e_meas[n_ready];
+    ephemeris_t ephe_cache[MAX_CHANNELS];
```

Comment:
I think `ephe_cache` should be static and the arrays of pointers should be on the stack.